### DISCUSSION TITLE
Do stores inherit from other stores? Can they?

### DISCUSSION THREAD

**Discussion by matthew-dean** [pmndrs/jotai#1450] [General] [answered]
Say I want to use the future (`unstable_createStore` / `createStore`) store definition. And say I have an app level store provider, but also want a kind of contextual provider to hold other values. So given this code:
```jsx
import { atom, createStore, Provider } from 'jotai'
import { Section } from './Section'

const globalStore = createStore()
export const appAtom = atom('foo')
globalStore.set(appAtom, 'bar')

const App = () => (
  <Provider store={globalStore}>
    <Page />
  </Provider>
)

// Section.jsx
import { createStore } from 'jotai'
import { Page } from './Page'

const contextStore = createStore()
export const localAtom = atom('one')
contextStore.set(localAtom, 'two')

export const Section = () => (
  <Provider store={contextStore}>
    <Page />
  </Provider>
)

// Page.tsx
import { useAtom } from 'jotai'
import { globalAtom } from './App'
import { localAtom } from './Section'

export const Page = () => {
  const [local] = useAtom(localAtom)
  const [glob] = useAtom(globalAtom)
  return (
    <div />
  ) 
}
```
In Page.tsx, my expectation is that `local` has a value of `'two'` and that `glob` has a value of `'bar'`. In other words, my expectation would be that because `globalAtom` does not have a value in the store, that the Provider would look in the store higher in the hierarchy of the tree to find a set value. Otherwise global atoms that have a set value in the app would effectively have their values "erased" by virtue of using another store.

If this doesn't work this way, could it?

**Comment by dai-shi** [comment, marked as answer]
It's technically impossible to inherit stores, because atoms can be defined dynamic. (For pre-defined atoms, it might be possible, but our design decision is not to do it at core because it complicates things. We may explore some possibilities for third-party libraries in the future.)

For your case, Provider `scope` is available. It's not a silver bullet, though.

```jsx
const App = () => (
  <Provider ... scope="a">
    <Provider ... scope="b">
      ...
    </Provider>
  </Provider>
)

const Page = () => {
  const [local] = useAtom(localAtom, 'b')
  const [glob] = useAtom(globalAtom, 'a')
  return (
    <div />
  ) 
}
```

**Reply by Lesik** [reply]
What would be the equivalent replacement for this functionality in v2? I don't believe the `scope` prop still exists.

**Reply by dai-shi** [reply]
That's still relevant if you follow the scope migration.
However, since then, I learned a lot about the use case, and for most of the cases, what people want isn't multiple stores. In such cases, what's recommended is atoms-in-atom or `jotai-molecules`.

**Comment by dai-shi** [comment]
I just developed another util for this kind of use cases. Hope someone can try.
https://github.com/jotaijs/jotai-scope/pull/2

**Comment by AjaxSolutions** [comment]
@dai-shi Why a Map is used (as opposed to a WeakMap) inside createScope?

**Reply by dai-shi** [reply]
Because it inherits from the parent map. `map = new Map(parentMap)`.
